pilot/pkg/networking/plugin/authz/rbac_v2.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authz converts Istio RBAC (role-based-access-control) policies (ServiceRole and AuthorizationPolicy)
+// to corresponding filter config that is used by the envoy RBAC filter to enforce access control to
+// the service co-located with envoy.
+// Currently the config is only generated for sidecar node on inbound HTTP/TCP listener. The generation
+// is controlled by RbacConfig (a singleton custom resource with cluster scope). User could disable
+// this plugin by either deleting the ClusterRbacConfig or set the ClusterRbacConfig.mode to OFF.
+// Note: ClusterRbacConfig is not created with default istio installation which means this plugin doesn't
+// generate any RBAC config by default.
+package authz
+
+import (
+	http_config "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/rbac/v2"
+	policyproto "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2alpha"
+
+	rbacproto "istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+// convertRbacRulesToFilterConfigV2 is the successor of convertRbacRulesToFilterConfig, which supports
+// converting AuthorizationPolicy.
+func convertRbacRulesToFilterConfigV2(service *serviceMetadata, option rbacOption) *http_config.RBAC {
+	rbac := &policyproto.RBAC{
+		Action:   policyproto.RBAC_ALLOW,
+		Policies: map[string]*policyproto.Policy{},
+	}
+
+	if option.globalPermissiveMode {
+		// Permissive Mode will be ignored.
+		// TODO(pitlv2109): Handle permissive mode in the future.
+		return &http_config.RBAC{Rules: rbac}
+	}
+
+	namespace := service.attributes[attrDestNamespace]
+
+	// Get all AuthorizationPolicy Istio config from this namespace.
+	allAuthzPolicies := option.authzPolicies.NamespaceToAuthorizationConfigV2[namespace].AuthzPolicies
+	// Get all key-value pairs of AuthorizationPolicy name to list of Subjects/ServiceRoleBindings in this namespace.
+	authzPolicyToSubjects := option.authzPolicies.AuthzPolicyToAllowSubjects(namespace)
+	for _, authzPolicy := range allAuthzPolicies {
+		// If a WorkloadSelector is used in the AuthorizationPolicy config and does not match this service, skip this
+		// AuthorizationPolicy.
+		if authzPolicy.Policy.WorkloadSelector != nil && !service.isServiceMatchedWorkload(authzPolicy.Policy.WorkloadSelector.Labels) {
+			continue
+		}
+		for i, subject := range authzPolicyToSubjects[authzPolicy.Name] {
```

Comment:
Why do we need `authzPolicyToSubjects[authzPolicy.Name]`?
Can we just use `authzPolicy.Policy.Allow` as each authzPolicy already has a list of its bindings.

Reply:
You're right. That was silly of me. Thanks :D